iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/AbstractBranch.java

```diff
@@ -80,54 +82,71 @@ public Side getSide(Terminal terminal) {
-                limits2 = limits;
-                notifyUpdate("currentLimits2", oldValue2, limits2);
-                break;
-            default:
-                throw new AssertionError("Unexpected Branch.Side value: " + side);
-        }
+    public List<OperationalLimits> getOperationalLimits1() {
+        return operationalLimitsHolder1.getOperationalLimits();
     }
 
     @Override
-    public CurrentLimits getCurrentLimits(Side side) {
-        switch (side) {
-            case ONE:
-                return limits1;
-            case TWO:
-                return limits2;
-            default:
-                throw new AssertionError("Unexpected Branch.Side value: " + side);
-        }
+    public <L extends OperationalLimits> L getOperationalLimits1(LimitType limitType, Class<L> limitClazz) {
+        return operationalLimitsHolder1.getOperationalLimits(limitType, limitClazz);
+    }
+
+    @Override
+    public CurrentLimitsAdder newCurrentLimits1() {
+        return operationalLimitsHolder1.newCurrentLimits();
+    }
 
+    @Override
+    public ApparentPowerLimitsAdder newApparentPowerLimits1() {
+        return operationalLimitsHolder1.newApparentPowerLimits();
     }
 
     @Override
-    public CurrentLimits getCurrentLimits1() {
-        return limits1;
+    public ActivePowerLimitsAdder newActivePowerLimits1() {
+        return operationalLimitsHolder1.newActivePowerLimitsAdder();
     }
 
     @Override
-    public CurrentLimitsAdder newCurrentLimits1() {
-        return new CurrentLimitsAdderImpl<>(Branch.Side.ONE, this);
+    public VoltageLimitsAdder newVoltageLimits1() {
+        return operationalLimitsHolder1.newVoltageLimits();
     }
 
     @Override
-    public CurrentLimits getCurrentLimits2() {
-        return limits2;
+    public List<OperationalLimits> getOperationalLimits2() {
+        return operationalLimitsHolder2.getOperationalLimits();
+    }
+
+    @Override
+    public <L extends OperationalLimits> L getOperationalLimits2(LimitType limitType, Class<L> limitClazz) {
+        return operationalLimitsHolder2.getOperationalLimits(limitType, limitClazz);
     }
 
     @Override
     public CurrentLimitsAdder newCurrentLimits2() {
-        return new CurrentLimitsAdderImpl<>(Branch.Side.TWO, this);
+        return operationalLimitsHolder2.newCurrentLimits();
+    }
+
+    @Override
+    public ApparentPowerLimitsAdder newApparentPowerLimits2() {
+        return operationalLimitsHolder2.newApparentPowerLimits();
+    }
+
+    @Override
+    public ActivePowerLimitsAdder newActivePowerLimits2() {
+        return operationalLimitsHolder2.newActivePowerLimitsAdder();
+    }
+
+    @Override
+    public VoltageLimitsAdder newVoltageLimits2() {
+        return operationalLimitsHolder2.newVoltageLimits();
+    }
+
+    OperationalLimitsHolderImpl getLimitsHolder1() {
+        return operationalLimitsHolder1;
+    }
+
+    OperationalLimitsHolderImpl getLimitsHolder2() {
+        return operationalLimitsHolder2;
```

Comment:
Is it really useful to have these getters?

Reply:
Yes, for topological merge in `NetworkImpl`